feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingsViewModel.kt

```diff
@@ -103,6 +101,93 @@ class SettingsViewModel @Inject constructor(
 //
 // ////////////////////////////////////////////////////////////
 
+    private fun getFlashUiState(
+        cameraAppSettings: CameraAppSettings,
+        constraints: SystemConstraints
+    ): FlashUiState {
+        val currentSupportedFlashModes =
+            constraints.forCurrentLens(cameraAppSettings)?.supportedFlashModes ?: emptySet()
+
+        check(currentSupportedFlashModes.isNotEmpty()) {
+            "No flash modes supported. Should at least support OFF."
+        }
+        val deviceSupportedFlashModes: Set<FlashMode> = constraints.forDevice(
+            CameraConstraints::supportedFlashModes
+        )
+        // disable entire setting when:
+        //  device only supports off... device unsupported rationale
+        //  lens only supports off... lens unsupported rationale
+        if (deviceSupportedFlashModes.size == 1) {
```

Comment:
nit: to be more precise with this `if` condition, use:
`if (deviceSupportedFlashModes == setOf(FlashMode.OFF))`

You can use this same check below for `currentSupportedFlashModes`.